pkg/virt-handler/vm.go

```diff
@@ -2561,6 +2564,72 @@ func (d *VirtualMachineController) vmUpdateHelperMigrationTarget(origVMI *v1.Vir
 	}
 	return nil
 }
+
+func ConfigureHousekeepingCgroup(vmi *v1.VirtualMachineInstance, d *VirtualMachineController) error {
+	cgroupManager, err := cgroup.NewManagerFromVM(vmi)
+	if err != nil {
+		return err
+	}
+
+	exists, err := cgroupManager.CreateChildCgroup("housekeeping", "cpuset")
+	// bail out if cgroup already configured
+	if exists == true {
+		return nil
+	}
+	if err != nil {
+		log.Log.Reason(err).Error("CreateChildCgroup ")
+		return err
+	}
+
+	key := controller.VirtualMachineInstanceKey(vmi)
+	domain, _, _, err := d.getDomainFromCache(key)
+	if err != nil {
+		return err
+	}
+
+	hkcpu, err := strconv.Atoi(domain.Spec.CPUTune.EmulatorPin.CPUSet)
```

Comment:
AdjustDomainForTopologyAndCPUSet() {
...
   	if vmi.Spec.Domain.CPU.IsolateEmulatorThread {
                emulatorThread, err = cpuPool.FitThread()
                if err != nil {
                        e := fmt.Errorf("no CPU allocated for the
emulation thread: %v", err)
                        log.Log.Reason(e).Error("failed to format
emulation thread pin")
                      	return e
                }                                                                                                                                                                               
                appendDomainEmulatorThreadPin(domain, emulatorThread)
        }                                                                                                                                                                                       
...
}                                                                                                                                                                                               

func appendDomainEmulatorThreadPin(domain *api.Domain, allocatedCpu uint32) {
        emulatorThread := api.CPUEmulatorPin{
                CPUSet: strconv.Itoa(int(allocatedCpu)),
        }                                                                                                                                                                                       
        domain.Spec.CPUTune.EmulatorPin = &emulatorThread
}                                                                                                                                                                                               

And:

    	if vmi.IsCPUDedicated() && vmi.Spec.Domain.CPU.IsolateEmulatorThread {
                err = configureHousekeepingCgroup(vmi, d)
                if err != nil {
                        return err
                }                                                                                                                                                                               
        }                                                                                                                                                                                       

So don't think it can be null.

Reply:
I don't think this guarantees that `domain.Spec.CPUTune` and `domain.Spec.CPUTune.EmulatorPin` are never nil.
In any case I would check it here to be safe.

Reply:
Fixed.